internal/bminventory/inventory_v2_handlers.go

```diff
@@ -750,6 +750,11 @@ func (b *bareMetalInventory) GetInfraEnvDownloadURL(ctx context.Context, params
 }
 
 func (b *bareMetalInventory) generateImageDownloadURL(ctx context.Context, infraEnvID, imageType, version, arch, imageTokenKey string) (string, *strfmt.DateTime, error) {
+	// MGMT-13668 - TODO: Remove this when https://issues.redhat.com/browse/RFE-3647 is resolved
+	if imageType != "full-iso" && (arch == common.PowerCPUArchitecture || arch == common.S390xCPUArchitecture) {
+		b.log.Warn("Image type minimal-iso unsupported for architecture %s, forcing full-iso", arch)
+		imageType = "full-iso"
```

Comment:
Should we communicate this in some better way?

Ideally we could fail the infraenv update/create in the REST API.

Not so sure about kube API. Is there such a thing as a "warning condition/status"?

Reply:
Actually thinking about this more ... we definitely shouldn't allow this combination at all in the REST case. We know the inputs are invalid there and the user can simply choose the correct option.

Reply:
KubeAPI, I reckon we should reject the invalid combination if we are going down this path. No warning, just don't allow it.

Reply:
Is there an ETA for it to be fixed?
If it will not happen anytime soon we can change block it in the api level. 
For kube-api it will be a sync error so it's the same

Reply:
Actually kube-api will not work because there is no option to select the image type so it will be done in a controller level

Reply:
I agree.  We shouldn't return something other than what the user requested.
For REST, we should fail the create/update, as @carbonin said.  In the worst case of the user modifying the URL to change the image type, the download should return 400.